src/test/groovy/org/prebid/server/functional/tests/pricefloors/PriceFloorsCurrencySpec.groovy

```diff
@@ -386,10 +387,43 @@ class PriceFloorsCurrencySpec extends PriceFloorsBaseSpec {
                 ["imp[].ext.prebid.floors.floorMinCur and ext.prebid.floors.floorMinCur has different values"]
 
         and: "Bidder request should contain floorMinCur, floorMin from request"
-        def bidderRequest = bidder.getBidderRequests(bidRequest.id).last()
-        assert bidderRequest.imp[0].ext.prebid.floors.floorMinCur == EUR
-        assert bidderRequest.imp[0].ext.prebid.floors.floorMin == FLOOR_MIN
-        assert bidderRequest.ext.prebid.floors.floorMinCur == JPY
-        assert bidderRequest.ext.prebid.floors.floorMin == FLOOR_MIN
+        verifyAll(bidder.getBidderRequest(bidRequest.id)){
+            imp[0].ext.prebid.floors.floorMinCur == EUR
+            imp[0].ext.prebid.floors.floorMin == FLOOR_MIN
+            ext.prebid.floors.floorMinCur == JPY
+            ext.prebid.floors.floorMin == FLOOR_MIN
+        }
+    }
+
+      def "PBS should replace floorValue on floorMin when floorMin biggest that floorValue defined in request"() {
+        given: "Default BidRequest with floorMin, floorMinCur, floorValue"
+        def randomFloorValue = PBSUtils.getRandomPrice(FLOOR_MIN as Integer)
+        def bidRequest = bidRequestWithFloors.tap {
+                imp[0].ext.prebid.floors = new ImpExtPrebidFloors().tap {
+                floorMinCur = USD
+                floorMin = randomFloorValue
+                floorValue = FLOOR_MIN
+            }
+        }
+
+        and: "Set Floors Provider response with a floorMin"
+        def floorsResponse = PriceFloorData.priceFloorData.tap {
+            modelGroups[0].values = [(rule): FLOOR_MIN]
+        }
+        floorsProvider.setResponse(bidRequest.site.publisher.id, floorsResponse)
```

Comment:
For a first request, we don't have floors data we have only a request to start fetching provider values
1) You can pass rules in request to use them from first call
2) The best case request.floorMin =2USD, imp.floorMin =4USD, modelGroup = 3USD (Resolved 4 for bidder)